### ISSUE TITLE
The word vector obtained by the word2vec tutorial is very bad

### ISSUE THREAD

**Issue by howl-anderson** [tensorflow/tensorflow#50645]
## URL(s) with the issue:

https://www.tensorflow.org/tutorials/text/word2vec

## Description of issue (what needs changing):

I run the word2vec code (without change) from the tutorial. But the word vector from the model is very bad, not reflect the semantic meaning at all.

Here is the result shown in the embedding projector:
![Selection_002](https://user-images.githubusercontent.com/1263428/124710517-7be51600-df2f-11eb-9dfb-7efd80bedeb6.png)

**Comment by UsharaniPagadala** [comment]
@howl-anderson 

Please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new?assignees=&labels=type%3Abug&template=00-bug-issue.md)

Could you please share the code/colab gist that you have implemented.Thanks

**Comment by howl-anderson** [comment]
@UsharaniPagadala 
I don't think it is a kind of bug of tensorflow, it more like a hyperparameter setting issue.

> Could you please share the code/colab gist that you have implemented.Thanks

I  use the jupyter notebook (https://colab.research.google.com/github/tensorflow/docs/blob/master/site/en/tutorials/text/word2vec.ipynb) provided by the tutorial (https://www.tensorflow.org/tutorials/text/word2vec). It will reproduce the result (not reflect the semantic meaning) in the embedding projector.

**Comment by UsharaniPagadala** [comment]
@howl-anderson 

Could please open a [stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow)/ Tensorflow discussion [forum](https://discuss.tensorflow.org/) issue as this is not a bug or feat request and move this to closed status.

**Comment by howl-anderson** [comment]
As I said before, it is not a bug of TensorFlow, it is a "bug" of the tutorial (something wrong with hyperparameter setting or corpus). I was being told (from tensorflow/docs) if there is an issue in docs, go here to report it. From the users' view, this tutorial is wrong. it is not a bug caused by the user.

**Comment by ymodak** [comment]
<img width="373" alt="Screen Shot 2021-07-09 at 11 16 52 AM" src="https://user-images.githubusercontent.com/42785357/125120336-2d5a8780-e0a7-11eb-980e-b7551ec30819.png">


I see relatively valid results. Can you please try again?
http://projector.tensorflow.org/

**Comment by howl-anderson** [comment]
Hi @ymodak, Thank your reply. But what you show in the previous comment is not the word embedding view. It is the view that the projector helps you (using edit distance or something) to choose what word do you want to query. It is NOT the query result. After you choose the query word, then it will give some nearest words with scores or distances. Also, Please also notice that we are discussing word embedding from the tutorial at https://www.tensorflow.org/tutorials/text/word2vec. In the tutorial, it using the corpus from Shakespeare's writing. I don't think Shakespeare knows the word "batman", which is the one of words in your previous image. It is more likely you are using the default embedding from the http://projector.tensorflow.org/.

**Comment by howl-anderson** [comment]
Also, I have already run the model more than 5 times, every time it gives meanless embedding.

**Comment by Aiman-Shivani** [comment]
Is this issue solved in any way? Because even i am getting the same issue of bad word embeddings after trying a lot

**Comment by Aiman-Shivani** [comment]
@howl-anderson Did you find any solution regarding this? If so please share
@MarkDaoust Could you please help us regarding this, have tried everything around the code but nothing works

**Comment by MarkDaoust** [comment]
@enigmaeth

Afroz you're the original author of this doc, do you have any ideas?

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
